contracts/core/Registry.sol

```diff
@@ -1,47 +1,174 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.0;
 
-// use Solady Roles for ownership of projects: https://github.com/Vectorized/solady/blob/main/src/auth/OwnableRoles.sol
-contract Registry {
+import { AccessControl } from "../../lib/openzeppelin-contracts/contracts/access/AccessControl.sol";
+
+import { Metadata } from "./libraries/Metadata.sol";
+
+contract Registry is AccessControl {
+
+    error NO_ACCESS_TO_ROLE();
+    error NOT_SUPPORTED();
+
+    /// @notice Unique identifier for an identity
+    uint private identityId;
+
+    /// @notice Types of roles assigned to an identity
+    enum RoleType {
+        OWNER,
+        MEMBER
+    }
+
+    /// @notice Struct to hold details of an identity
     struct IdentityDetails {
         string name;
-        string metadata; // @todo update to MetaPtr
+        Metadata metadata;
         address attestationAddr;
```

Comment:
Maybe we should call this `identity` instead of `attestationAddr`?

Or really just something that doesn't have `attestation` or `addr` in the name because:
1. The type is `address`, so we don't need to have the type in the variable name too
2. Attestations are just on of the things these addresses can be used for - SBTs is another

Thinking through the naming:

Conceptually, the "identity" is the combination of name, metadata, and address. So we'll want to do something like:

```solidity
identity.name;
identity.metadata;
identity.<something>;
```

Some ideas:
- location / locator
- resource
- wallet
- identifier (though not a fan of `identity.identifier`)
- recipient

Reply:
I agree with @0xZakk on this, I am leaning toward `ownerIdentifier` though 🤩

```javascript
struct IdentityDetails {
    string name;
    Metadata metadata;
    address ownerIdentifier;
}
```

WDYT?

Reply:
The core concept makes sense to me. I like the idea of having the name convey that it's central to identity.

Riffing on that:
- location / locator (from your original list)
- anchor
- nucleus
- locus
- kernel
- hub

Reply:
anchor is sticking for me ⚓️

Reply:
anchor works.